### PR TITLE
Fixed: Large content responses crash trace logger causing hung thread.

### PR DESCRIPTION
**Pull request by psylenced**
#### Database Migration
NO

#### Description
When trace logging is enabled and the HttpClient makes a request with a large content response (eg. Usenet history), the trace log call which dumps the response results in a hung thread.

I had this occur with a response length of approx 700k.

There were no exceptions thrown, the thread simply hung and was unable to recover.

### CODE REVIEW

File: src/NzbDrone.Common/Http/HttpClient.cs [hunk 1]
```diff
@@ -136,7 +136,16 @@ private HttpResponse ExecuteRequest(HttpRequest request, CookieContainer cookieC
 
             if (request.LogResponseContent)
             {
-                _logger.Trace("Response content ({0} bytes): {1}", response.ResponseData.Length, response.Content);
+                const int MaxContentLength = 50_000;
```
Comment (psylenced) [inline src/NzbDrone.Common/Http/HttpClient.cs:139]: Need some advice as to what the max length should be. (50, 100?).

I found 700~ was enough to hang the thread.

**Comment by austinwbest** [comment]
I am not going to agree with this change but that doesnt mean others wont. This is not a common thing, i can not even think of another instance on discord of someone mentioning it. With that said, forcing a change across the board for everyone because one user had an issue one time doesn't seem quite like the way to go about it. If you know of another person, no need to link conversations and what not, the point remains the same where it is not common.

There could possibly be some config.xml entry or something that allows people with these kind of issues to add a value? Something a little more configurable and optional basically would be my thought on this instead of a blanket change.

Just my 2 cents, not worth much of course so take it with a grain of salt :)

**Comment by RobinDadswell** [comment]
What version are you running @psylenced. We had a similar issue with syslogging recently but this is now resolved. That issue was reproduceable but I don't believe we have had anyone else call this out as an issue before so it could be local system related. 

TLDR I am leaning towards this not being changed unless we can prove that it's an issue outside of the one system you are experiencing this issue on.

**Comment by psylenced** [comment]
I'm running this on the latest develop build. I was trying to debug another issue and came across this.

The 2 side effects I observed was:

1. Application wouldn't start - just hung. (I believe history() is run on startup).

2. When I went into the database and put a invalid domain name in, the app would start. When I corrected that via the UI and attempted to save the donwload client - the app hung again.

As far as I'm aware, ```request.LogResponseContent``` is only true for trace logs isn't it? So hopefully shouldnt affect too many people. 

I'm happy to investigate further and find a specific length of response which causes this.

From what I can gather, anyone who has a semi-large nzbget history who runs in trace mode will be affected by this issue.

The primary reason I put this through was due to it not actually throwing an exception. Meaning the app will hang and the only way out is to disable trace mode.

**Comment by psylenced** [comment]
https://github.com/psylenced/Radarr/tree/hung-thread-issue

**Comment by psylenced** [comment]
On further testing, even 20k in log size is taking a good 20+ seconds per request on a 32 thread cpu.

A response of 10-20k on any HttpClient request is very easily reachable by a end-user in trace mode.

**Comment by psylenced** [comment]
Have done some further investigation and have tracked it down to a single RegEx which is run to cleanse the log files before they are written.

https://github.com/Radarr/Radarr/blob/9b21408a03c137b635ecf4c4534a9c0bd9cdb1b4/src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs#L21

Log:
```
F:/Projects/TestCleanse/bin/Debug/net6.0/TestCleanse.exe 
00.06 - (?<=\?|&|: )(apikey|(?:access[-_]?)?token|passkey|auth|authkey|user|uid|api|[a-z_]*apikey|account|passwd)=(?<secret>[^&=]+?)(?=[ "&=]|$)
00.08 - (?<=\?|&)[^=]*?(username|password)=(?<secret>[^&=]+?)(?= |&|$)
00.09 - torrentleech\.org/(?!rss)(?<secret>[0-9a-z]+)
00.09 - torrentleech\.org/rss/download/[0-9]+/(?<secret>[0-9a-z]+)
00.09 - iptorrents\.com/[/a-z0-9?&;]*?(?:[?&;](u|tp)=(?<secret>[^&=;]+?))+(?= |;|&|$)
00.09 - /fetch/[a-z0-9]{32}/(?<secret>[a-z0-9]{32})
00.09 - getnzb.*?(?<=\?|&)(r)=(?<secret>[^&=]+?)(?= |&|$)
00.09 - (?<=[?& ;])[^=]*?(_?(?<!use|get_)token|username|passwo?rd)=(?<secret>[^&=]+?)(?= |&|$|;)
536.78 - announce(\.php)?(/|%2f|%3fpasskey%3d)(?<secret>[a-z0-9]{16,})|(?<secret>[a-z0-9]{16,})(/|%2f)announce
536.78 - C:\\Users\\(?<secret>[^\"]+?)(\\|$)
536.78 - /home/(?<secret>[^/"]+?)(/|$)
536.79 - "Name"\s*:\s*"[^"]*(username|password)"\s*,\s*"Value"\s*:\s*"(?<secret>[^"]+?)"
536.79 - "[^"]*(username|password|api_?key|nzb_key)"\s*:\s*"(?<secret>[^"]+?)"
536.79 - "email_(account|to|from|pwd)"\s*:\s*"(?<secret>[^"]+?)"
536.79 - \["[a-z._]*(username|password)",\d,"(?<secret>[^"]+?)"
536.79 - \["(boss_key|boss_key_salt|proxy\.proxy)",\d,"(?<secret>[^"]+?)"
536.79 - auth.login\("(?<secret>[^"]+?)"
536.79 - "?method"?\s*:\s*"(getTorrents)",\s*"?params"?\s*:\s*\[\s*"(?<secret>[^"]+?)"
536.79 - getTorrents\("(?<secret>[^"]+?)"
536.79 - (?<=\?|&)(authkey|torrent_pass)=(?<secret>[^&=]+?)(?="|&|$)
536.79 - (?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)
536.79 - api/v[0-9]/notification/radarr/(?<secret>[\w-]+)
```

The first set of regex all completed in 0.03 seconds, the single expression in reference above took 536 seconds, the rest took 0.01 seconds.

**Comment by psylenced** [comment]
Possible fix added. Need some feedback in relation to what was being replaced to make sure it works correctly.

Definitely was related to prowler change:

https://github.com/Radarr/Radarr/pull/6873#pullrequestreview-929604048

**Review by RobinDadswell — APPROVED** [review]

**Comment by snakefoot** [comment]
Note after upgrading to NLog5, then NLog FileTarget was fixed to actual enable the cleanse:

https://github.com/Radarr/Radarr/pull/7365/files#diff-0a7a3850a41818c5e0032d1267cbf4ae62dd5f27b7c7b6d8cd26755d0dae38d6

The `GetFormattedMessage` stopped being called with [NLog v4.4.2](https://github.com/NLog/NLog/releases/tag/v4.4.2), and now with the upgrade to NLog v5 the cleanse was enabled again in Radarr.
